### ISSUE TITLE
Reorganization of files

### ISSUE THREAD

**Issue by jmlarson1** [Libensemble/libensemble#285]
We have resources files:
````
libensemble/env_resources.py
libensemble/mpi_resources.py
libensemble/node_resources.py
libensemble/resources.py
````
controller files
````
libensemble/balsam_controller.py
libensemble/controller.py
libensemble/mpi_controller.py
````
two util locations
````
libensemble/util/
libensemble/utils.py
````
as well as
````
libensemble/tests/regression_tests/common.py
libensemble/tests/regression_tests/support.py
libensemble/gen_funcs/support.py
libensemble/alloc_funcs/support.py
````

Do we have a coherent motivation for this structure?

**Comment by shuds13** [comment]
I've been thinking about sub-dirs for resources and controllers. This will change the import API though (for controllers). Bear in mind we are thinking of renmaing that also, so we might want to make that change in one go!

Currently the util/ directory is, I think, internal modules rather than user API. We might want to consider that.

**Comment by jlnav** [comment]
Further considerations from Slack discussion:
 - [ ]  Have two utils directories; one for libE-intended functions, one for external user-intended convenience functions
 - [ ] Reformat internal routines to start with underscores, autodoc

**Comment by jlnav** [comment]
Closed by #344
